exporter/exporterhelper/common.go

```diff
@@ -106,9 +107,16 @@ func WithTimeout(timeoutSettings TimeoutSettings) Option {
 
 // WithRetry overrides the default RetrySettings for an exporter.
 // The default RetrySettings is to disable retries.
-func WithRetry(retrySettings RetrySettings) Option {
+func WithRetry(config RetrySettings) Option {
 	return func(o *baseExporter) {
-		o.retrySender = newRetrySender(o.set.ID, retrySettings, o.set.Logger, o.onTemporaryFailure)
+		if !config.Enabled {
+			o.retrySender = &errorLoggingRequestSender{
+				logger:  o.set.Logger,
+				message: "Exporting failed. Dropping data. Try enabling retries to survive temporary failures.",
```

Comment:
It's not necessary dropping data. If the batch processor is not enabled, the error can be propagated to the receiver and can be retried

Reply:
I did not change the messages, as you can see they are previous messages when component were disabled. We can revisit them in a separate PR>